### ISSUE TITLE
Add validation and training sets to NNPredictor

### ISSUE THREAD

**Issue by JBlaschke** [ML4FnP/GMRES-Learning#12]
Use validation data to test the "tolerance" of the forward model. We can also use the training loss function to check if the NNPredictor needs more gradient descent steps.
